complete.py

```diff
@@ -0,0 +1,771 @@
+            context_cache : openmmtools.cache.ContextCache, optional
+                The ContextCache to use for Context creation. If None, the global cache
+                openmmtools.cache.global_context_cache is used (default is None).
+            reassign_velocities : bool, optional
+                If True, the velocities will be reassigned from the Maxwell-Boltzmann
+                distribution at the beginning of the move (default is False).
+            n_restart_attempts : int, optional
+                When greater than 0, if after the integration there are NaNs in energies,
+                the move will restart. When the integrator has a random component, this
+                may help recovering. On the last attempt, the ``Context`` is
+                re-initialized in a slower process, but better than the simulation
+                crashing. An IntegratorMoveError is raised after the given number of
+                attempts if there are still NaNs. 
+            reporter : coddiwomple.openmm.reporter.OpenMMReporter, default None
+                a reporter object to write trajectories
+        """
+        super().__init__(openmm_pdf_state,
+                 integrator,
+                 context_cache,
+                 reassign_velocities,
+                 n_restart_attempts)
+        #create a pdf state for the subset indices (usually a vacuum system)
+        self.pdf_state_subset = openmm_pdf_state_subset
+        assert self.pdf_state_subset.temperature == self.pdf_state.temperature, f"the temperatures of the pdf states do not match"
+        
+        #create a dictionary for subset indices
+        self._subset_indices_map = subset_indices_map
+        
+        #create an ani handler attribute that can be referenced
+        self.ani_handler = ani_handler
+        
+        #create a context for the subset atoms that can be referenced
+        self.context_subset, _ = cache.global_context_cache.get_context(self.pdf_state_subset)
+        
+        #create a reporter for the accumulated works
+        self._state_works = {}
+        self._state_works_counter = 0
+        
+        #create a reporter
+        self._write_trajectory = False if reporter is None else True
+        self.reporter=reporter
+        if self._write_trajectory:
+            from coddiwomple.particles import Particle
+            self.particle = Particle(0)
+        else:
+            self.particle = None
+        
+    def _before_integration(self, *args, **kwargs):
+        particle_state = args[0] #define the particle state
+        n_iterations = args[1] #define the number of iterations
+        
+        self._current_state_works = [] #define an interim (auxiliary) list that will track the thermodynamic work of the current application
+        self._current_state_works.append(0.0) #the first incremental work is always 0 since the importance function is identical to the first target distribution (i.e. fully interacting MM)
```

Comment:
this convention is off by one from what I would expect -- is this typical? want to recover simple importance sampling when the number of intermediates is zero.